.github/workflows/linea-devnet.yml

```diff
@@ -0,0 +1,117 @@
+name: linea-devnet
+
+on:
+  workflow_call:
+    outputs:
+      workflow_run_id: 
+        description: "the run id of the workflow"
+        value: ${{ jobs.build.outputs.workflow_run_id }}
+  workflow_dispatch:
+
+permissions:
+  id-token: write  
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    environment: build
+    outputs:
+      workflow_run_id: ${{ steps.workflow_details.outputs.id }}
+    steps:
+
+      - name: checkout
+        uses: actions/checkout@v4
+
+      - name: get workflow_details
+        id: workflow_details
+        run: echo "id=${{ github.run_id }}" >> $GITHUB_OUTPUT
+      
+      - name: get versions via dotenv
+        id: dotenv
+        uses: falti/dotenv-action@v1
+        with:
+          path: versions/linea-devnet.env
+          mode: development
+          keys-case: lower
+          log-variables: true
+          load-mode: strict
+
+      - name: set docker tag
+        id: dockertag
+        run: |
+          if [ "${{ github.event_name }}" == "push" ] && [ "${{ github.ref_type }}" == "tag" ]; then
+            echo "IMAGE=devnet-${{ github.ref_name }}" >> $GITHUB_OUTPUT
+          else
+            echo "IMAGE=devnet-$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
+          fi
+
+      - name: download and untar the linea-besu archive
+        run: |
+          cd /tmp/
+          echo "downloading linea-besu: ${{ steps.dotenv.outputs.linea_besu_tar_gz }}"
+          wget -nv "https://artifacts.consensys.net/public/linea-besu/raw/names/linea-besu.tar.gz/versions/${{ steps.dotenv.outputs.linea_besu_tar_gz }}/linea-besu-${{ steps.dotenv.outputs.linea_besu_tar_gz }}.tar.gz"
+          tar -xvf linea-besu-${{ steps.dotenv.outputs.linea_besu_tar_gz }}.tar.gz
+          mkdir -p /opt/besu
+          mv /tmp/linea-besu-${{ steps.dotenv.outputs.linea_besu_tar_gz }}/* /opt/besu/
+          echo "moved to /opt/besu"
+
+      - name: copy the list of versions to the container
+        run: |
+          cp ./versions/linea-devnet.env /opt/besu/versions.txt
+          mkdir -p /opt/besu/plugins
+
+      - name: get the plugins
```

Comment:
My understanding is that this logic used to be done be gradle.build script. In order to make a release what needs to be done by GHA and what will be done by the gradle.build? What is the reason to move this logic from gradle.build here?